mobile/src/screens/journal/journal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import PlusIcon from '#assets/img/icons/plus.svg';
+import {
+  Card,
+  Header,
+  InputSearch,
+  LinearGradient,
+  Link,
+  ScrollView,
+  View,
+} from '#libs/components/components';
+import { AppColor } from '#libs/enums/enums';
+import {
+  useAppDispatch,
+  useAppRoute,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useNavigation,
+  useSearch,
+} from '#libs/hooks/hooks';
+import { actions as journalActions } from '#slices/journal/journal';
+
+import { styles } from './styles';
+
+const Journal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation();
+  const route = useAppRoute();
+
+  const { allJournalEntries } = useAppSelector(({ journal }) => {
+    return {
+      allJournalEntries: journal.allJournalEntries,
+    };
+  });
+  const badgeCount = allJournalEntries.length;
+
+  const { filteredData: filteredJournals, setSearchQuery } = useSearch(
+    allJournalEntries,
+    'title',
+  );
+
+  const handleSelectJournal = useCallback(() => {
+    // TODO: Implement actual functionality for the onPress event
+  }, []);
+
+  useEffect(() => {
+    navigation.setOptions({
+      header: () => {
+        return <Header title={route.name} badgeCount={badgeCount} />;
+      },
+    });
+  }, [navigation, route.name, badgeCount]);
+
+  useEffect(() => {
+    void dispatch(journalActions.getAllJournalEntries());
+  }, [dispatch]);
+
+  return (
+    <LinearGradient>
+      <View style={styles.container}>
+        <InputSearch
+          placeholder="Search journal"
+          setSearchQuery={setSearchQuery}
+        />
+        <ScrollView contentContainerStyle={styles.list}>
+          {filteredJournals.map((item) => {
+            return (
+              <Card
+                title={item.title}
+                onPress={handleSelectJournal}
+                key={item.id}
+              />
+            );
+          })}
+        </ScrollView>
+        <View style={styles.linkWrapper}>
+          <Link
```

Comment:
Maybe it would be Button?

Reply:
Can I change the type of the label from string to React.ReactNode in the Button component to pass an icon? I could pass the icon separately as a prop, but I already have such an implementation in my other task 'Add Meditation Player Layout', with completely different logic. Or should I go back to using a different Button component in 'Add Meditation Player Layout' as it was before and then pass the icon as a prop here?

Reply:
let's leave it for now, maybe we will change this in future